packages/app/src/specs/SpecsListBanners.vue

```diff
@@ -249,32 +222,33 @@ watch(
 )
 
 const cloudData = computed(() => ([props.gql.cloudViewer, props.gql.cachedUser, props.gql.currentProject] as const))
+const bannerToShow = computed(() => {
+  const componentsByStatus = {
+    isLoggedOut: LoginBanner,
+    needsOrgConnect: CreateOrganizationBanner,
+    needsProjectConnect: ConnectProjectBanner,
+    needsRecordedRun: RecordBanner,
+  }
 
-watch(
-  cloudData,
-  ([cloudViewer, cachedUser, currentProject]) => {
-    // Cached user covers state where we're authenticated but data isn't loaded yet
-    const isLoggedIn = !!cachedUser?.id || !!cloudViewer?.id
-    // Need to be able to tell whether the lack of `firstOrganization` means they don't have an org or whether it just hasn't loaded yet
-    // Not having this check can cause a brief flicker of the 'Create Org' banner while org data is loading
-    const isOrganizationLoaded = !!cloudViewer?.firstOrganization
-    const isMemberOfOrganization = (cloudViewer?.firstOrganization?.nodes?.length ?? 0) > 0
-    const isProjectConnected = !!currentProject?.projectId && currentProject.cloudProject?.__typename === 'CloudProject'
-    const hasNoRecordedRuns = currentProject?.cloudProject?.__typename === 'CloudProject' && (currentProject.cloudProject?.runs?.nodes?.length ?? 0) === 0
-    const hasFourDaysOfCypressUse = (Date.now() - currentProject?.savedState?.firstOpened) > interval('4 days')
+  return {
+    component: componentsByStatus[loginConnectStore.userStatus] ?? null,
+    cohortOption: getCohortForBanner(bannerIds[loginConnectStore.userStatus]),
+  }
```

Comment:
```suggestion
  return reactive({
    component: componentsByStatus[loginConnectStore.userStatus] ?? null,
    cohortOption: getCohortForBanner(bannerIds[loginConnectStore.userStatus]),
  })
```
Wrapping the return in `reactive` should allow for the un-nesting of the `refs` but still allow them to be reactive.  That way, all the nested `.value` calls can be removed.  I have not fully tested this yet, but I believe this should work.

Reply:
nice, I will check that out, I had drafted a comment about this but your review beat me to it 🙏